1calculator.c

```diff
@@ -0,0 +1,215 @@
+
+int is_space(char ch) {
+    if(ch == ' ') return 1;
+    return 0;
+}
+
+int is_digit(char ch) {
+    if(ch >= '0' && ch <= '9')return 1;
+    return 0;
+}
+
+// Definition for operator stack
+typedef struct
+{
+    int top;
+    char data[MAX_SIZE];
+} operator;
+
+// Definition for operand stack
+typedef struct
+{
+    int top;
+    int data[MAX_SIZE];
+} operand;
+
+// Funtion associated with operator stack
+void push_operator(operator* stk, char val)
+{
+    // Error handling
+    if (stk->top == MAX_SIZE - 1)
+    {
+        printf("Error: stack overflow\n");
+        exit(EXIT_FAILURE);
+    }
+    stk->top++;
+    stk->data[stk->top] = val;
+}
+
+void pop_operator(operator* stk)
+{
+    // Error handling
+    if (stk->top == -1)
+    {
+        printf("Error: stack underflow\n");
+        exit(EXIT_FAILURE);
+    }
+
+    stk->top--;
+}
+
+// Funtion associated with operand stack
+void push_operand(operand *stk, int val)
+{
+    // Error handling
+    if (stk->top == MAX_SIZE - 1)
+    {
+        printf("Error: stack overflow\n");
+        exit(EXIT_FAILURE);
+    }
+
+    stk->top++;
+    stk->data[stk->top] = val;
+}
+
+void pop_operand(operand *stk)
+{
+    // Error handling
+    if (stk->top == -1)
+    {
+        printf("Error: stack underflow\n");
+        exit(EXIT_FAILURE);
+    }
+    stk->top--;
+}
+
+// Operator precedance checking function
+int precedance(char operator)
+{
+    // Associativity of + & - being same
+    if (operator== '+' || operator== '-')
+        return 0;
+
+    // Associativity of * & / being same
+    if (operator== '*' || operator== '/')
+        return 1;
+}
+
+// Function for solving binary operations
+void solve_operator(operand * or, operator* op)
+{
+    // Extracting variables from operand stack
+    int num1 = or->data[or->top];
+    pop_operand(or);
+    int num2 = or->data[or->top];
+    pop_operand(or);
+
+    // Extracting operator from operator stack
+    char operator= op->data[op->top];
+    pop_operator(op);
+
+    // Result evaluation
+    int result;
+    switch (operator)
+    {
+    case '+':
+        result = num2 + num1;
+        break;
+
+    case '-':
+        result = num2 - num1;
+        break;
+
+    case '*':
+        result = num2 * num1;
+        break;
+
+    case '/':
+
+        // Checking division by 0 condition
+        if (num1 == 0)
+        {
+            printf("Error: Division by zero");
+            exit(EXIT_FAILURE);
+        }
+
+        result = num2 / num1;
+        break;
+    }
+
+    // Pushing result in operand stack
+    push_operand(or, result);
+}
+
+// Function for evaluating the expression
+int evaluate_expression(char *expression)
+{
+
+    // Declaring operator & operand stack
+    operator operatorStack;
+    operatorStack.top = -1;
+    operand operandStack;
+    operandStack.top = -1;
+
+    // Parsing the expression
+    int ind;
+    for ( ind = 0; expression[ind] != '\0'; ind++)
+    {
+        char ch = expression[ind];
+
```

Comment:
The variable name 'ch' is unclear and does not convey its purpose or meaning in the context of the code.
Give Proper meaningfull name.